can't build scrypt in Archlinux

With :

```
 scrypt git:(master) ✗ gcc --version
gcc (GCC) 6.1.1 20160802
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

ldd (GNU libc) 2.24
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```
and ruby-2.3.1, I can't build this gem.


```
 scrypt git:(master) ✗ rake compile_ffi
gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -fPIC   -o ext/scrypt/x86_64-linux/scryptenc_cpuperf.o -c ext/scrypt/scryptenc_cpuperf.c
ext/scrypt/scryptenc_cpuperf.c:105:21: warning: ‘struct timespec’ declared inside parameter list will not be visible outside of this definition or declaration
 getclocktime(struct timespec * ts)
                     ^~~~~~~~
ext/scrypt/scryptenc_cpuperf.c: In function ‘getclocktime’:
ext/scrypt/scryptenc_cpuperf.c:111:4: error: dereferencing pointer to incomplete type ‘struct timespec’
  ts->tv_sec = tv.tv_sec;
    ^~
ext/scrypt/scryptenc_cpuperf.c: At top level:
ext/scrypt/scryptenc_cpuperf.c:119:21: warning: ‘struct timespec’ declared inside parameter list will not be visible outside of this definition or declaration
 getclockdiff(struct timespec * st, double * diffd)
                     ^~~~~~~~
ext/scrypt/scryptenc_cpuperf.c: In function ‘getclockdiff’:
ext/scrypt/scryptenc_cpuperf.c:121:18: error: storage size of ‘en’ isn’t known
  struct timespec en;
                  ^~
ext/scrypt/scryptenc_cpuperf.c:125:27: error: dereferencing pointer to incomplete type ‘struct timespec’
  *diffd = (en.tv_nsec - st->tv_nsec) * 0.000000001 +
                           ^~
ext/scrypt/scryptenc_cpuperf.c:121:18: warning: unused variable ‘en’ [-Wunused-variable]
  struct timespec en;
                  ^~
ext/scrypt/scryptenc_cpuperf.c: In function ‘scryptenc_cpuperf’:
ext/scrypt/scryptenc_cpuperf.c:139:18: error: storage size of ‘st’ isn’t known
  struct timespec st;
                  ^~
ext/scrypt/scryptenc_cpuperf.c:139:18: warning: unused variable ‘st’ [-Wunused-variable]
rake aborted!
Command failed with status (1): [gcc -fexceptions -O -fno-omit-frame-pointe...]
/home/zerosign/.rvm/gems/ruby-2.3.1@scrypt/gems/ffi-compiler-1.0.1/lib/ffi-compiler/compile_task.rb:133:in `block (2 levels) in define_task!'
/home/zerosign/.rvm/gems/ruby-2.3.1@scrypt/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/zerosign/.rvm/gems/ruby-2.3.1@scrypt/bin/ruby_executable_hooks:15:in `eval'
/home/zerosign/.rvm/gems/ruby-2.3.1@scrypt/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => compile_ffi => ffi-compiler:default => ext/scrypt/x86_64-linux/libscrypt_ext.so => ext/scrypt/x86_64-linux/scryptenc_cpuperf.o
(See full trace by running task with --trace)
```

It seems that either current *libc header* are being altered.

I'm a very similar compilation error on ruby-2.4 / gcc 6.3.1:

```gcc -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -D__need_timespec -fPIC   -o x86_64-linux/scryptenc_cpuperf.o -c ./scryptenc_cpuperf.c
./scryptenc_cpuperf.c:105:21: warning: ‘struct timespec’ declared inside parameter list will not be visible outside of this definition or declaration
 getclocktime(struct timespec * ts)
                     ^~~~~~~~
./scryptenc_cpuperf.c: In function ‘getclocktime’:
./scryptenc_cpuperf.c:111:4: error: dereferencing pointer to incomplete type ‘struct timespec’
  ts->tv_sec = tv.tv_sec;
    ^~
./scryptenc_cpuperf.c: At top level:
./scryptenc_cpuperf.c:119:21: warning: ‘struct timespec’ declared inside parameter list will not be visible outside of this definition or declaration
 getclockdiff(struct timespec * st, double * diffd)
                     ^~~~~~~~
./scryptenc_cpuperf.c: In function ‘getclockdiff’:
./scryptenc_cpuperf.c:121:18: error: storage size of ‘en’ isn’t known
  struct timespec en;
                  ^~
./scryptenc_cpuperf.c:125:27: error: dereferencing pointer to incomplete type ‘struct timespec’
  *diffd = (en.tv_nsec - st->tv_nsec) * 0.000000001 +
                           ^~
./scryptenc_cpuperf.c:121:18: warning: unused variable ‘en’ [-Wunused-variable]
  struct timespec en;
                  ^~
./scryptenc_cpuperf.c: In function ‘scryptenc_cpuperf’:
./scryptenc_cpuperf.c:139:18: error: storage size of ‘st’ isn’t known
  struct timespec st;
                  ^~
./scryptenc_cpuperf.c:139:18: warning: unused variable ‘st’ [-Wunused-variable]
rake aborted!
Command failed with status (1): [gcc -fexceptions -O -fno-omit-frame-pointe...]
/home/cameron/.gem/ruby/2.4.0/gems/ffi-compiler-1.0.1/lib/ffi-compiler/compile_task.rb:133:in `block (2 levels) in define_task!'
Tasks: TOP => default => x86_64-linux/libscrypt_ext.so => x86_64-linux/scryptenc_cpuperf.o
(See full trace by running task with --trace)

rake failed, exit code 1
```
As you can see, the D__need_timespec flag is present.  I can make this thing build by changing -std=c99 to -std=gnu99 (specs still pass), but I'm not sure if that's an acceptable solution

--- a/Rakefile
+++ b/Rakefile
@@ -27,7 +27,7 @@ namespace "ffi-compiler" do
     t.cflags << "-Wall -std=c99"
     t.cflags << "-msse -msse2" if t.platform.arch.include? "86"
     t.cflags << "-D_GNU_SOURCE=1" if RbConfig::CONFIG["host_os"].downcase =~ /mingw/
-    t.cflags << "-D__need_timespec" if RbConfig::CONFIG['host_os'].downcase =~ /linux/
+    t.cflags << "-D_POSIX_C_SOURCE=199309L" if RbConfig::CONFIG['host_os'].downcase =~ /linux/
     t.cflags << "-arch x86_64 -arch i386" if t.platform.mac?
     t.ldflags << "-arch x86_64 -arch i386" if t.platform.mac?
 

--- a/ext/scrypt/Rakefile
+++ b/ext/scrypt/Rakefile
@@ -4,7 +4,7 @@ FFI::Compiler::CompileTask.new('scrypt_ext') do |t|
   t.cflags << "-Wall -std=c99"
   t.cflags << "-msse -msse2" if t.platform.arch.include? "86"
   t.cflags << "-D_GNU_SOURCE=1" if RbConfig::CONFIG["host_os"].downcase =~ /mingw/
-  t.cflags << "-D__need_timespec" if RbConfig::CONFIG['host_os'].downcase =~ /linux/
+  t.cflags << "-D_POSIX_C_SOURCE=199309L" if RbConfig::CONFIG['host_os'].downcase =~ /linux/
   t.cflags << "-arch x86_64 -arch i386" if t.platform.mac?
   t.ldflags << "-arch x86_64 -arch i386" if t.platform.mac?
   t.export '../../lib/scrypt/scrypt_ext.rb'

